### ISSUE TITLE
Magento 2.3.5 - Configurable product showing all associate product in configuration grid

### ISSUE THREAD

**Issue by iovistainc** [magento/magento2#28967]
### Preconditions (*)

1. Magento Version:2.3.5-p1
2. MySQL 5.7

### Steps to reproduce (*)

1. Create Configurable Product and add more then 60+ associate simple/virtual products.
2. Edit newly created Configurable Product and go to the Configuration section 
3. Click on the next page to see other products.
4. You will see all the product in second page instead of total of 20 product

### Expected result (*)

1. After click on the next page button, another 20 product will be display in the grid.

### Actual result (*)

1. It loads all the remaining associate products even if per page size it set 20 products per page.
2. It makes so many loads on the server that even sometimes the page was crashed.

![aFz5k](https://user-images.githubusercontent.com/16300151/86321179-16908200-bc56-11ea-9d41-dff1f6a6a35f.png)

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Additional information**:
Please refer this [comment](https://github.com/magento/magento2/issues/28967#issuecomment-652880150)

**Comment by swathishs** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @swathishs. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @swathishs, here is your [Magento instance](https://i-28967-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-28967-2-4-develop.instances.magento-community.engineering/admin_8c82](https://i-28967-2-4-develop.instances.magento-community.engineering/admin_8c82)
Login: `7ee3859e` Password: `66d2c49418a1`
Instance will be terminated in up to 3 hours.

**Comment by swathishs** [comment]
@iovistainc : Issue is not reproducible in Magento 2.4 develop instance 
Elastic cache 7+
Please find the attachment below:
![kKi9JkeNb7](https://user-images.githubusercontent.com/58288685/86326356-37110a00-bc5f-11ea-8896-d986b8517370.gif)

**Comment by swathishs** [comment]
@magento give me 2.3.5-p1 instance

**Comment by magento-engcom-team** [comment]
Hi @swathishs. Thank you for your request. I'm working on Magento 2.3.5-p1 instance for you

**Comment by iovistainc** [comment]
@swathishs Issue is also available on Magento 2.3.3.

**Comment by swathishs** [comment]
@iovistainc  : You can check in this instance -here is your Magento instance.
Admin access: https://i-28966-2-3-5-p1.instances.magento-community.engineering/admin_edf1
Login: 74fbe3e0 Password: 9b84f4fa71ab
Instance will be terminated in up to 3 hours.

The issue is not reproducible in Magento 2.3.5-p1 instance
![image](https://user-images.githubusercontent.com/58288685/86334289-a809ef00-bc6a-11ea-81a2-ef160d59c290.png)

![JSxyNX0tLZ](https://user-images.githubusercontent.com/58288685/86334654-22d30a00-bc6b-11ea-8d59-f73db63c048e.gif)

**Comment by iovistainc** [comment]
@swathishs I have checked the Magento 2.3.5-p1 instance and the issue is there.
Once you open the product grid will show 20 Products but when you click on the next page it display all product but it should have display 20 product as we have set 20 products per page. You can count the product once you click on the next page. It will load 50 products.

**Comment by swathishs** [comment]
Did you check in above instance

**Comment by iovistainc** [comment]
Yes, I have checked in above 2.3.5-p1 instance. You can also count the product on the next page of the grid.

**Comment by swathishs** [comment]
Could send me the config product URL of the product and put some screenshot or video

**Comment by iovistainc** [comment]
@swathishs 
Config product SKU: new
Here is the screencast: https://www.loom.com/share/f9cbd31c460c4bf888b716a2fc04c1a3

**Comment by swathishs** [comment]
Sorry I have just clicked the next arrow but didn't see the total product. 
The issue is reproducible in 2.3.5p1
![dDlLGkecAa](https://user-images.githubusercontent.com/58288685/86338217-ad1d6d00-bc6f-11ea-9833-cc4b36f3037a.gif)

**Comment by iovistainc** [comment]
@swathishs  Is it reproducible for you on 2.3.5-p1 or not?

**Comment by swathishs** [comment]
Ya it's reproducible in 2.3.5-p1

**Comment by swathishs** [comment]
The issue can be reproduced only on 2.3.5p1 but not on 2.4 develop instance

**Comment by iovistainc** [comment]
@swathishs  Could you please help me with how to solve this issue for a 2.3.5-p1 instance?

**Comment by iovistainc** [comment]
This issue can be reproduced on 2.3.3 also.

**Comment by swathishs** [comment]
@iovistainc : Sorry I am tester I validate the ticket and approve whether the issue is recreated on Magento develop instance.

**Comment by magento-engcom-team** [comment]
:x: Something went wrong. Cannot update Jira issue (https://jira.corp.magento.com/browse/MC-35592).

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @swathishs 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-35592` were created

**Issue Available**: @swathishs, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
:x: Something went wrong. Cannot update Jira issue (https://jira.corp.magento.com/browse/MC-35592).

**Comment by magento-engcom-team** [comment]
:x: Something went wrong. Cannot update Jira issue (https://jira.corp.magento.com/browse/MC-35592).

**Comment by magento-engcom-team** [comment]
:x: Something went wrong. Cannot update Jira issue (https://jira.corp.magento.com/browse/MC-35592).

**Comment by magento-engcom-team** [comment]
:x: Something went wrong. Cannot update Jira issue (https://jira.corp.magento.com/browse/MC-35592).

**Comment by magento-engcom-team** [comment]
:x: Something went wrong. Cannot update Jira issue (https://jira.corp.magento.com/browse/MC-35592).

**Comment by iovistainc** [comment]
This commit has fixed the issue.
https://github.com/magento/magento2/commit/76493a97c457fbf368e5ee53dcf62a7d64437f2d

**Comment by sdzhepa** [comment]
Hello @iovistainc 
fyi: @swathishs 

After review in the internal Jira I found next
1. This issue has been fixed for the 2.3.x release line in the scope of `MC-33502`. The fix has been tested and delivered into `2.3-develop` branch (not available on a public, its private/internal branch). Target release line for this fix is **2.3.6**
2. The fix has been ported for 2.3.x release line in the scope of `MC-33523`. The fix has been tested and delivered into `2.4-develop` branch (available on a public https://github.com/magento/magento2/commit/76493a97c457fbf368e5ee53dcf62a7d64437f2d). Target release line for this fix is **2.4.1**

So, I am closing this issue as fixed.  Need just to wait for 2.3.6 to have it in release.
Also can be used to commit from this comment https://github.com/magento/magento2/issues/28967#issuecomment-655386861 to porting it.
